lms/security.py

```diff
@@ -199,6 +208,49 @@ def permits(self, request, _context, permission):
         return _permits(self.identity(request), permission)
 
 
+class EmailSettingsSecurityPolicy:
+    def __init__(self, email_unsubscribe_service: EmailUnsubscribeService):
+        self.helper = AuthTktCookieHelper(
+            secret="secret",
+            cookie_name="email.settings",
+            secure=False,
+            timeout=60 * 5,
+            reissue_time=0,
+            max_age=60 * 5,
+            path="/email/settings",
+            http_only=True,
+            wild_domain=False,
+            parent_domain=False,
+            domain="localhost",
+        )
+        self.email_unsubscribe_service = email_unsubscribe_service
+
+    def identity(self, request):
+        if token := request.params.get("token"):
+            try:
+                return self.email_unsubscribe_service.decode_token(token)["h_userid"]
+            except InvalidJWTError:
+                return None
+
+        try:
+            return self.helper.identify(request)["userid"]
+        except (KeyError, TypeError):
+            return None
+
+    def authenticated_userid(self, request):
+        return self.identity(request)
+
+    def permits(self, request, _context, permission):
+        identity = self.identity(request)
+        if identity is not None and permission == "email.settings":
```

Comment:
We should include the permission string on the Enum in this file `Permissions(Enum)`